### PR TITLE
Update module workload visualization 

### PR DESCRIPTION
**Pull request by ZhangYiJiang**
This is part of #426 

This stablizes the design and code for module workload visualization. This should handle (most) workloads well, without any major glitches.

`outline: 1px solid` is added to all screenshots below to to show the HTML structure. 

Half and quarter hour workloads are now handled using vertical blocks. This does mean that in situations like the one below, a workload which should technically fit on one row (10 hrs) would not. This edge case happens infrequently enough that I don't think it's worth the amount of work to fix it. 

screenshot from 2017-09-13 16-35-43

The component no longer resizes blocks so that they always fill up the entire horizontal space. This makes block width consistent across different modules. 

screenshot from 2017-09-13 16-33-52

Component header now shows the number of hours if there is enough space (5 hrs+). This reduces the need to count the blocks manually, which would be difficult for components with a lot of hours. 

screenshot from 2017-09-13 16-33-19

A high workload module which would have been laid out incorrectly on the old code. 

screenshot from 2017-09-13 16-33-37

### Behind the scene changes 

- Split the two component SCSS for `module-workload.scss` and `timeslot-table.scss` out from `module-finder.scss`
- Unfortunately `$FlowFixMe` has to be applied to a case where we use `_.entries`, because the return type for that function is annotated incorrectly

### CODE REVIEW

File: v3/src/js/views/components/module-info/ModuleWorkload.jsx [hunk 1]
```diff
@@ -18,84 +19,109 @@ const shortComponentNames: { [WorkloadComponent]: string } = {
...
-            </span>}
-            <div className="module-workload-blocks">
-              {_.range(count).map(i => (
-                <div key={i} className={`workload-${component.toLowerCase()}-bg`} />
-              ))}
+  render() {
+    const workload = parseWorkload(this.props.workload);
+    if (typeof workload === 'string') return this.renderFallback();
+
+    const total = _.sum(_.values(workload));
```
[the remark this answers is not in the document]
Comment (ZhangYiJiang) [reply v3/src/js/views/components/module-info/ModuleWorkload.jsx]: Hmmm? This isn't counting the number of workload components, but summing up the workload components to get the total number of hours

**Comment by ZhangYiJiang** [comment]
Problem with using the slanted half approach is that there are quarter hour workloads. Yeah, I know... 

This sort of works-ish for any decimal fraction (it might be hard to tell the difference between 0.1 and 0.15, but I think 1, 0.5 and 0.25 are distinguishable enough).

**Comment by ZhangYiJiang** [comment]
@li-kai Updated: 

screenshot from 2017-09-14 11-50-42

(There's actually a module with 0.7hrs workload lol)

**Comment by yangshun** [comment]
What are the advantages of showing 9 blocks for 9 hours vs just putting 9 hours directly?

**Comment by ZhangYiJiang** [comment]
In the most common use case, modules will have 10 hrs of workload (4MC), so you get 10 blocks on a single row. The advantage of using this representation is that

- It takes up a little less space
- It's easy to see how much time is taken up by each component proportionally
  (eg. this module is mostly lab, this module requires more time on projects, etc.)

(These are for the most common case of 10-20 hrs per week.)

**Comment by yangshun** [comment]
I see. Ok then (:

**Pull request merged**